benchmark/checks/check_mem_usage.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Memory usage check."""
+import os
+import time
+from threading import Thread
+
+from benchmark.checks.utils import (
+    SyncedGeneratorConnection,
+    make_agent,
+    make_envelope,
+    make_skill,
+    wait_for_condition,
+)
+
+import click
+
+import psutil  # type: ignore
+
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.skills.base import Handler
+
+
+class TestHandler(Handler):
+    """Dummy handler to handle messages."""
+
+    SUPPORTED_PROTOCOL = DefaultMessage.protocol_id
+
+    def setup(self) -> None:
+        """Noop setup."""
+
+    def teardown(self) -> None:
+        """Noop teardown."""
+
+    def handle(self, message: Message) -> None:
+        """Handle incoming message."""
+        self.context.outbox.put(make_envelope(message.to, message.sender))
+
+
+@click.command()
+@click.option("--duration", default=3, help="Run time in seconds.")
+@click.option(
+    "--runtime_mode", default="async", help="Runtime mode: async or threaded."
+)
+def main(duration, runtime_mode):
```

Comment:
We need to run main multiple times to be able to measure the variance in the measurements as well. I'd start with 100 times.